probe-rs/src/probe/jlink/mod.rs

```diff
@@ -276,6 +276,21 @@ impl ProbeFactory for JLinkFactory {
     }
 }
 
+fn requires_connection_handle(selector: &DebugProbeSelector) -> bool {
```

Comment:
Can we not detect whether registration is in the featureset?

Reply:
I don't know how, it's unclear which devices need it, which ones support it, and which ones die if you try. It's all a mess mostly created by cheap clones so I'm not hopeful we can do this is a nicer way.

Reply:
J-Links have a feature detection with a flag system and I was hoping it as mapped there, but I guess it's really just an issue with clones not caring.

Reply:
Well we _are_ checking for the `Register` capability bit so yeah 🤷‍♂️